partiql-ast/src/main/pig/partiql.ion

```diff
@@ -802,55 +802,83 @@ may then be further optimized by selecting better implementations of each operat
             )
         )
 
+        (with statement
```

Comment:
Also need an entry in the CHANGELOG for the plan modeling changes and `PartiQLResult` changes.